ophyd/status.py

```diff
@@ -264,12 +264,35 @@ def __init__(self, positioner, target, *, start_ts=None,
         self.pos = positioner
         self.target = target
         self.start_ts = start_ts
+        self.start_pos = self.pos.position
         self.finish_ts = None
         self.finish_pos = None
 
+        self._watchers = []
+
         # call the base class
         super().__init__(positioner, **kwargs)
 
+    def watch(self, func):
+        """
+        Subscribe to notifications about progress. Useful for progress bars.
+
+        Parameters
+        ----------
+        func : callable
+            Expected signature: ``func(status_obj, fraction_done)``
+        """
+        self._watchers.append(func)
+
+    def notify_watchers(self):
+        try:
+            fraction = ((self.pos.position - self.start_pos) /
+                        (self.target - self.start_pos))
+        except Exception as exc:
+            fraction = exc
```

Comment:
I am concerned that the subtraction operation above might fail in some situations, and I don't want that to kill a run in progress -- at least until this feature has seen battle-tested on the floor. I thought it might sense to report the exception back to the watcher. In JavaScript callback conventionally have the signature `f(result, err)`, a pattern I thought about adopting here. But in Python we can check `isinstance(res, Exception)` and be on our way.